src/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -11,26 +11,53 @@
 
 namespace Microsoft.Identity.Client.Instance.Discovery
 {
+    /// <summary>
+    /// Priority order of metadata providers: 
+    /// 
+    /// 1. Static cache (this is populated from the network)
+    /// 2. Well-known cache if all enviroments present in the token cache are known (this is hardcoded into msal)
+    /// 3. Cache stored in token cache (Not currently implemented)
+    /// 4. User provided AAD instance metadata 
```

Comment:
Add `<see cref="AbstractApplicationBuilder.WithInstanceDiscoveryMetadata"/>` ?

Reply:
Ok, but pls note that this isn't public API.